### PR TITLE
feat(perf issues): Add type and category to serializer

### PR DESCRIPTION
**Pull request by ceorourke**
Add the issue's type and category to the `GroupSerializerBase`.

### CODE REVIEW

File: src/sentry/api/serializers/models/group.py [hunk 1]
```diff
@@ -329,6 +329,8 @@ def serialize(
             "subscriptionDetails": subscription_details,
             "hasSeen": attrs["has_seen"],
             "annotations": attrs["annotations"],
+            "issueType": obj.issue_type.name.lower(),
+            "issueCategory": obj.issue_category.name.lower(),
```
Comment (ceorourke) [inline src/sentry/api/serializers/models/group.py:333]: I wasn't sure if we wanted `group.Type`, `groupType`, `issue.Type`, etc. so I just went with this.

**Review by wedamija — APPROVED** [review]

**Review by gggritso — APPROVED** [review]
Makes sense!

FWIW I really like the idea of `"issue": {"category": "performance", "type": "n_plus_one"}` instead of `issueType` and `issueCategory`! I think it's clearer and it matches the search syntax `"issue.type:<x>"` and `"issue.category:<y>"` which is nice.

P.S.1. Maybe confusing to have `"issue.category"` be `"performance"` but `type` be `"transaction"`. Should they match? Do they ever not line up that way?
P.S.2. There's a funny case where `type`, `issue.category,` and `issue.type` are _all_ `"error"`. Nothing wrong, just seems a little confusing. `undefined` would be fine for error groups, too
P.S.3. If only `type` was called `eventType`! Maybe once this PR lands, we can remove usage of `type` on the frontend

**Comment by wedamija** [comment]
>FWIW I really like the idea of "issue": {"category": "performance", "type": "n_plus_one"} instead of issueType and issueCategory! I think it's clearer and it matches the search syntax "issue.type:<x>" and "issue.category:<y>" which is nice.
I'm not totally sure if this makes sense, since `Group` and `Issue` are used interchangeably, so it's kind of like saying `issue["issue"]["performance"]`. Doesn't have to be a blocker though

> P.S.1. Maybe confusing to have "issue.category" be "performance" but type be "transaction". Should they match? Do they ever not line up that way?
We don't want them to match - not necessarily a guarantee that `performance` will always be backed by a transaction, or that transactions will only ever create `performance` issues. Maybe we decide there's another category of issues we can determine via transactions?

> P.S.3. If only type was called eventType! Maybe once this PR lands, we can remove usage of type on the frontend
I'm not sure that we can remove uses of `type` here. It has a totally separate meaning and use case from `issueType`.

**Review by wedamija — APPROVED** [review]

**Comment by gggritso** [comment]
@wedamija 👌🏻 fair enough

**Pull request merged**